larndsim/lightLUT.py

```diff
@@ -0,0 +1,119 @@
+
+    (lut_min,lut_max,lut_ndiv) = lut_geometry
+    vox_xyz = np.floor(pos/(lut_max-lut_min)*lut_ndiv).astype(int)+lut_ndiv/2
+    voxel = vox_xyz[2]*lut_ndiv[0]*lut_ndiv[1]+vox_xyz[1]*lut_ndiv[0]+vox_xyz[0]
+
+    return voxel
+
+
+def GetHalfDetCopy(pos): # Determines which TPC the edep takes place in
+    tpc_x = math.floor(pos[0]/(consts.ModuleDimension[0]/2.))+consts.n_mod[0]/2*consts.n_tpc[0]
+
+    return int(tpc_x)%2
+
+def LarndToLUTCoord(pos,lut_geometry): # Converts the LArND-sim coord. system to that of the LUT
+
+    # access LUT geometry
+    lut_min,lut_max,lut_ndiv = lut_geometry
+    
+    # should add another condition to be greater than whatever the thckness of the cathode is 
+    lut_pos = pos + np.array([lut_min[0],220,0])
+
+    return (lut_pos)
+
+########################################################################################################
+
+def calculate(t_data,lut_path,light_dat): # Simulates the number of photons read by each optical channel depending on ewhere the edep occurs. Also indicates the time it takes for a photon to reach the nearest photomultiplier tube (the "fastest" photon)
+    
+    # Loads in LUT root file
+    np_lut = np.load(lut_path)
+    
+    # Obtains lut geometry
+    lut_geometry = GetLutGeometry(lut_path)  
+    
+    # Data containers
+    time = np.full((t_data['dE'].size,consts.n_op_channel*2),20.)
+    tphotons = np.zeros((t_data['dE'].size,consts.n_op_channel*2))
+    
+    # Defines variables of global position
+    x = t_data['x'] # The average of the position between x_start and x_end from the edep-sim file
+    y = t_data['y']
+    z = t_data['z']
+
+    # Defining number of produced photons (from quencing.py)
+    n_photons = light_dat['n_photons_edep'][:,0]
+
+    # Loop edep positions
+    for dE in range(len(t_data['dE'])):
+
+        # Global position
+        pos = (np.array((z[dE],y[dE],x[dE])))*consts.cm2mm
+
+        sys.stdout.write('\r    current position: ' + str(pos) + ('(%.1f %%)' % ((dE+1)/float(len(t_data['dE']))*100)))
+        sys.stdout.flush()
+        
+        # tpc
+        tpc = GetHalfDetCopy(pos)
+        
+        # LUT position
+        lut_pos = LarndToLUTCoord(pos,lut_geometry) 
+
+        # voxel containing LUT position
+        voxel = GetVoxel(lut_pos,lut_geometry)
+        
+        # Calls voxel data
+        lut_vox = np_lut[np_lut['Voxel'] == voxel]
+        
+        op_dat = lut_vox['OpChannel']
+        vis_dat = lut_vox['Visibility']
+        T1_dat = lut_vox['T1']
+        
+        # loop voxel entry-list
+        for entry in range(len(lut_vox)):
+
+            op_channel = op_dat[entry]
+            
+            # Calculates the number of photons reaching each optical channel
+            n_photons_read = vis_dat[entry]*n_photons[dE]
+            
+            # Flips op channels if edep occurs in tpc 2
+            if (tpc==1):
```

Comment:
does this still work in a detector with more than one module?

Reply:
It will now!  TPC selection is done by reusing the `pixel_plane` variable set in `drifting.py`.  Light calculation step is moved to after this step and before pixel simulation.